krm-functions/lib/kptrl/kptrl.go

```diff
@@ -0,0 +1,133 @@
+/*
+ Copyright 2023 Nephio.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package kptrl
+
+import (
+	"sync"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+type ResourceList interface {
+	// AddResult add a result with error and corresponding kubeObject by
+	// appending a result to the result slice in the resourcelist
+	AddResult(err error, obj *fn.KubeObject)
+	// GetResults gets the results slice from the resourcelist
+	GetResults() fn.Results
+	// GetObject return an fn sdk KubeObject by comparing the APIVersion, Kind and Name
+	// if the object is found the corresponding obj is returned, if not nil is returned
+	GetObject(obj *fn.KubeObject) *fn.KubeObject
+	// GetObjects returns all items from the resourcelist
+	GetObjects() fn.KubeObjects
+	// SetObject sets the object in the resourcelist items. It either updates/overrides
+	// the entry if it exists or appends the entry if it does not exist in the resourcelist
+	// It uses APIVersion, Kind and Name to check the object uniqueness
+	SetObject(obj *fn.KubeObject)
+	// DeleteObject deletes the object from the resourcelist if it exists.
+	DeleteObject(obj *fn.KubeObject)
```

Comment:
@henderiw Based on the latest fn-runtime library Some functions are missing;
e.g: AddObject(o)

Can you please update this to latest?

Reply:
@henderiw Similarly `NewMutator` function is missing from the latest kptfile lib: https://github.com/nokia/nephio-nephio/tree/kptfile-lib

can you please add it.

Reply:
AddObject is set object and NewMutator is New